browser/ipfs/ipfs_tab_helper.cc

```diff
@@ -192,6 +194,17 @@ void IPFSTabHelper::DNSLinkResolved(const GURL& ipfs,
 #if !BUILDFLAG(IS_ANDROID)
       && !auto_redirect_blocked) {
     LoadUrlForAutoRedirect(GetIPFSResolvedURL());
+    auto ipfs_always_start_infobar_delegate_factory =
+        std::make_unique<BraveIPFSAlwaysStartInfoBarDelegateFactory>(
+            ipfs::IpfsServiceFactory::GetForContext(
+                web_contents()->GetBrowserContext()),
+            pref_service_);
+    if (disable_resolve_method_check_for_testing_) {
```

Comment:
I think we can move check if resolution method is local node to IpfsTabHelper so you can override this method is tests

Reply:
done